src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -67,18 +67,20 @@ type DataCache = Record<string, Promise<LoadDataReturnValue>>;
 // V3 dictionary helper functions
 function updateExpiredDepositsV3(dict: ExpiredDepositsToRefundV3, deposit: V3DepositWithBlock): void {
   // A deposit refund for a deposit is invalid if the depositor has a bytes32 address input for an EVM chain. It is valid otherwise.
-  if (chainIsEvm(deposit.originChainId) && !isValidEvmAddress(deposit.depositor)) {
+  if (chainIsEvm(deposit.originChainId) && !deposit.depositor.isValidEvmAddress()) {
```

Comment:
wdyt about internalising the chain -> address type mapping within the Address class, such that we could write: `!deposit.depositor.valid(deposit.originChainId)` ?

Reply:
Good call. Done here https://github.com/across-protocol/sdk/pull/904/commits/16dbd7e71e3f07706b0b708ea5e94aba544c84f5